### ISSUE TITLE
Support symbolic naming of resources

### ISSUE THREAD

**Issue by StephenWeatherford** [microsoft/vscode-azurearmtools#1308]

**Comment by MarcusFelling** [comment]
@alex-frankel - do you have a timeline for when this needs to get done?

**Comment by alex-frankel** [comment]
@anthony-c-martin -- the first thing we need symbolic name for is extensibility, yeah? If so, I'd say the soonest would be early CY22. I'm ok if we ship early previews of extensibility and arm tools doesn't support symbolic name right away.

**Comment by MarcusFelling** [comment]
We also need to cost out a smaller version of this as a contingency to only support a subset of the above.

**Comment by anthony-c-martin** [comment]
> @anthony-c-martin -- the first thing we need symbolic name for is extensibility, yeah? If so, I'd say the soonest would be

Correct. It was initially introduced as a way of making code generation simpler, but we don't technically **need** that.
